pkg/client/block/client.go

```diff
@@ -58,20 +73,80 @@ type blockClient struct {
 	// These enable the EventsReplayClient to correctly map the raw event bytes
 	// to Block objects and to correctly return a BlockReplayObservable
 	eventsReplayClient client.EventsReplayClient[client.Block]
+	// latestBlockMu is a mutex that protects the latestBlock field from concurrent
+	// writes from getLatestBlocks and getInitialBlock.
+	latestBlockMu *sync.Mutex
+	// latestBlock is the last block observed by the blockClient.
+	// It is initialized by the getInitialBlock function and then updated by the
+	// updateLatestBlock goroutine that listens to the eventsReplayClient.
+	latestBlock client.Block
 }
 
 // CommittedBlocksSequence returns a replay observable of new block events.
 func (b *blockClient) CommittedBlocksSequence(ctx context.Context) client.BlockReplayObservable {
 	return b.eventsReplayClient.EventsSequence(ctx)
 }
 
-// LatestsNBlocks returns the last n blocks observed by the BockClient.
+// LastNBlocks returns the last n blocks observed by the BlockClient.
 func (b *blockClient) LastNBlocks(ctx context.Context, n int) []client.Block {
 	return b.eventsReplayClient.LastNEvents(ctx, n)
 }
 
+// LastBlock returns the last blocks observed by the BlockClient.
+func (b *blockClient) LastBlock() client.Block {
+	b.latestBlockMu.Lock()
+	defer b.latestBlockMu.Unlock()
```

Comment:
This lock doesn't protect b.latestBlock against concurrent access.

We can either:
- Copy the block before return
- Since the return value is only used for getting the height and hash, we could maintain those as fields in the block client